osu.Game/Screens/Edit/Setup/MetadataSection.cs

```diff
@@ -3,79 +3,117 @@
+                    !string.IsNullOrEmpty(metadata.ArtistUnicode) ? metadata.ArtistUnicode : metadata.Artist),
+                RomanisedArtistTextBox = createTextBox<LabelledRomanisedTextBox>("Romanised Artist",
+                    !string.IsNullOrEmpty(metadata.Artist) ? metadata.Artist : MetadataUtils.StripNonRomanisedCharacters(metadata.ArtistUnicode)),
+
+                Empty(),
+
+                TitleTextBox = createTextBox<LabelledTextBox>("Title",
+                    !string.IsNullOrEmpty(metadata.TitleUnicode) ? metadata.TitleUnicode : metadata.Title),
+                RomanisedTitleTextBox = createTextBox<LabelledRomanisedTextBox>("Romanised Title",
+                    !string.IsNullOrEmpty(metadata.Title) ? metadata.Title : MetadataUtils.StripNonRomanisedCharacters(metadata.ArtistUnicode)),
+
+                Empty(),
+
+                creatorTextBox = createTextBox<LabelledTextBox>("Creator", metadata.AuthorString),
+                difficultyTextBox = createTextBox<LabelledTextBox>("Difficulty Name", Beatmap.BeatmapInfo.Version),
+                sourceTextBox = createTextBox<LabelledTextBox>("Source", metadata.Source),
+                tagsTextBox = createTextBox<LabelledTextBox>("Tags", metadata.Tags)
             };
 
             foreach (var item in Children.OfType<LabelledTextBox>())
                 item.OnCommit += onCommit;
         }
 
+        private TTextBox createTextBox<TTextBox>(string label, string initialValue)
+            where TTextBox : LabelledTextBox, new()
+            => new TTextBox
+            {
+                Label = label,
+                FixedLabelWidth = LABEL_WIDTH,
+                Current = { Value = initialValue },
+                TabbableContentContainer = this
+            };
+
         protected override void LoadComplete()
         {
             base.LoadComplete();
 
-            if (string.IsNullOrEmpty(artistTextBox.Current.Value))
-                GetContainingInputManager().ChangeFocus(artistTextBox);
+            if (string.IsNullOrEmpty(ArtistTextBox.Current.Value))
+                GetContainingInputManager().ChangeFocus(ArtistTextBox);
+
+            ArtistTextBox.Current.BindValueChanged(artist => transferIfRomanised(artist.NewValue, RomanisedArtistTextBox));
+            TitleTextBox.Current.BindValueChanged(title => transferIfRomanised(title.NewValue, RomanisedTitleTextBox));
+            updateReadOnlyState();
+        }
+
+        private void transferIfRomanised(string value, LabelledTextBox target)
+        {
+            if (MetadataUtils.IsRomanised(value))
+                target.Current.Value = value;
+
+            updateReadOnlyState();
+            updateMetadata();
+        }
+
+        private void updateReadOnlyState()
+        {
+            RomanisedArtistTextBox.ReadOnly = MetadataUtils.IsRomanised(ArtistTextBox.Current.Value);
+            RomanisedTitleTextBox.ReadOnly = MetadataUtils.IsRomanised(TitleTextBox.Current.Value);
```

Comment:
Cannot use `.Disabled` because of https://github.com/ppy/osu-framework/issues/3919, again. I stared some more at that issue again to see if I can prod it forward somehow, but couldn't conjure up anything that I haven't proposed in that thread already. That text layout schedule which is causing problems seems unremovable.